torch/_refs/__init__.py

```diff
@@ -614,6 +620,43 @@ def add(
 # TODO: add docstring
 # complex =  _make_elementwise_binary_reference(prims.complex, type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT)
 
+
+@elementwise_type_promotion_wrapper(
+    type_promoting_args=("a", "b"),
+    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT,
+)
+def _floor_divide(
```

Comment:
So the challenge with floor_divide is that in PyTorch floor_divide is incorrectly performing truncation division -- we plan to change this after the 1.12 branch cut (which should be very soon), though

Reply:
Aha! This PR correctly accounts for that, my mistake being so hasty